api/server/middleware_auth_test.go

```diff
@@ -0,0 +1,112 @@
+package server
+
+import (
+	"context"
+	"github.com/libopenstorage/openstorage/pkg/auth"
+	"github.com/libopenstorage/openstorage/pkg/auth/systemtoken"
+	"net/http"
+	"reflect"
+	"testing"
+)
+
+type mockAuthenticator struct {
+	claims *auth.Claims
+}
+
+func (a *mockAuthenticator) AuthenticateToken(context context.Context, token string) (*auth.Claims, error) {
+	return a.claims, nil
+}
+
+func (a *mockAuthenticator) Username(claims *auth.Claims) string {
+	return a.claims.Name
+}
+
+type noTokenGenerator struct{}
+
+func (n *noTokenGenerator) GetToken(opts *auth.Options) (string, error) {
+	return "", nil
+}
+
+// Issuer returns the token issuer for this generator necessary
+// for registering the authenticator in the SDK.
+func (n *noTokenGenerator) Issuer() string {
+	return ""
+}
+
+// GetAuthenticator returns an authenticator for this issuer used by the SDK
+func (n *noTokenGenerator) GetAuthenticator() (auth.Authenticator, error) {
+	return nil, nil
+}
+
+func TestNewSecurityMiddleware(t *testing.T) {
```

Comment:
@stgleb we should have tests for business logic coded in `securityHandler()`

Reply:
Done